base/gmp.jl

```diff
@@ -707,8 +707,16 @@ end
 
 factorial(x::BigInt) = isneg(x) ? BigInt(0) : MPZ.fac_ui(x)
 
-binomial(n::BigInt, k::UInt) = MPZ.bin_ui(n, k)
-binomial(n::BigInt, k::Integer) = k < 0 ? BigInt(0) : binomial(n, UInt(k))
+function binomial(n::BigInt, k::Integer)
+    k < 0 && return BigInt(0)
+    k <= typemax(Culong) && return binomial(n, Culong(k))
+    n < 0 && return isodd(k) ? -binomial(k - n - 1, k) : binomial(k - n - 1, k)
+    κ = n - k
+    κ < 0 && return BigInt(0)
+    κ <= typemax(Culong) && return binomial(n, Culong(κ))
+    throw(OverflowError("Computation will exceed memory"))
```

Comment:
I think it should be “would exceed” instead of “will exceed”, because the computation is not performed if the error is thrown.
```suggestion
    throw(OverflowError("Computation would exceed memory"))
```

Reply:
Very debatable; but if we really want this I will yield with no further convincing: just bear in mind that [gmp.jl:633](https://github.com/JuliaLang/julia/blob/648f6825c61992ce4ba0d91bb180528404daf280/base/gmp.jl#L633) already has the error message:
```
throw(OverflowError("exponent $y is too large and computation will overflow"))
``` 
not authored by me. So consistently it would be my judgement that using "will" would seem more appropriate than using would would seem to be.

Reply:
I would say that other place should also be adjusted then. Having existing bad phrasing should not be a reason for using it more?

Reply:
Ok, I will yield then. As for the "other place", please open another issue.